src/extension/appLauncher.ts

```diff
@@ -84,17 +93,16 @@ export class AppLauncher {
         }
     }
 
-    public openFileAtLocation(openFileRequest: OpenFileRequest): Promise<void> {
-        const { filename, lineNumber } = openFileRequest;
-        return new Promise((resolve) => {
+    public openFileAtLocation(filename: string, lineNumber: number): Q.Promise<void> {
+        return Q.Promise((resolve) => {
```

Comment:
This function was refactored, since its caller code requires Q.Promise type ([getReactNativeWorkspaceForFile](https://github.com/microsoft/vscode-react-native/blob/descriptor-factory-approach-refactoring/src/extension/openFileAtLocation.ts#L52) method). Refactoring of all the caller code and dependent parts could lead to significant changes through all the extension's code.